src/main/java/net/sf/jabref/logic/bibtexkeypattern/BibtexKeyPatternUtil.java

```diff
@@ -519,6 +521,13 @@ public static String applyModifiers(String label, List<String> parts, int offset
                     }
                     resultingLabel = abbreviateSB.toString();
 
+                } else if ("camel".equals(modifier)) {
+                    // CamelCasingOfEveryWord
+                    Title title = new Title(resultingLabel);
+                    title.getWords().stream().forEach(Word::toUpperFirst);
```

Comment:
If this code works, then it relies on a bad designed `Title` class. A collection exposed by a `get` method shouldn't give access to the internal state of the object. I think the best solution would be to add methods to the `Title` class which convert the title to a lower/upper/whatever case (i.e. `Title toLower()`). Since this probably goes beyond what you want to do in this PR, could you at least add a method `getWordsReadonly` (returning `Collections.unmodifiableList(words)`) and mark `getWords` deprecated. Thanks.